app/services/data_migrations/remove_surplus_reference_selections.rb

```diff
@@ -0,0 +1,25 @@
+module DataMigrations
+  class RemoveSurplusReferenceSelections
+    TIMESTAMP = 20210616160559
+    MANUAL_RUN = false
+
+    def change
+      selected_reference_counts =
+        ApplicationForm.joins(:application_references)
+        .where(references: { selected: true })
+        .group('id')
+        .select('"application_forms".id, COUNT("references".id) AS ref_count')
+
+      applications_with_surplus_selections =
+        ApplicationForm.where(
+          id: selected_reference_counts.select { |c| c.ref_count > 2 }.map(&:id),
+        )
```

Comment:
Can you replace this with `.having("references".id > 2)` on the query above? Would avoid loading selected reference counts for all the 'normal' applications.